src/Symfony/Component/Security/Core/Authentication/Token/UsernamePasswordToken.php

```diff
@@ -44,6 +46,8 @@ public function __construct($user, $credentials, $providerKey, array $roles = ar
         $this->providerKey = $providerKey;
 
         parent::setAuthenticated(count($roles) > 0);
+
+        $this->setShouldUpdateRoles($user instanceof UserInterface && $user->getRoles() === $roles);
```

Comment:
Is this is expected... this affects all tokens today no? Should we optin thru config instead?

Reply:
This was my way of trying to safely opt everyone in automatically... but I don't think that's possible anymore, thanks to https://github.com/symfony/symfony/pull/24331#issuecomment-332690665

So yea, I think we'll need to have a way to opt into this somehow...